resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DeletingRuntimeChecker.scala

```diff
@@ -0,0 +1,104 @@
+package com.broadinstitute.dsp
+package resourceValidator
+
+import cats.effect.Concurrent
+import cats.mtl.Ask
+import cats.syntax.all._
+import org.broadinstitute.dsde.workbench.google2.DataprocClusterName
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.util2.InstanceName
+import org.typelevel.log4cats.Logger
+
+// Looks for GCE, Dataproc, or Azure runtime that have been stuck in a `deleting` status for over 60 minutes
+object DeletingRuntimeChecker {
+  def impl[F[_]](
+    dbReader: DbReader[F],
+    deps: RuntimeCheckerDeps[F]
+  )(implicit F: Concurrent[F], logger: Logger[F], ev: Ask[F, TraceId]): CheckRunner[F, Runtime] =
+    new CheckRunner[F, Runtime] {
+      override def appName: String = resourceValidator.appName
+
+      override def configs = CheckRunnerConfigs(s"deleting-runtime", shouldAlert = true)
+
+      override def dependencies: CheckRunnerDeps[F] = deps.checkRunnerDeps
+
+      override def resourceToScan: fs2.Stream[F, Runtime] = dbReader.getDeletingRuntimes
+
+      override def checkResource(runtime: Runtime, isDryRun: Boolean)(implicit
+        ev: Ask[F, TraceId]
+      ): F[Option[Runtime]] =
+        runtime match {
+          case x: Runtime.Dataproc =>
+            checkDataprocCluster(x, isDryRun)
+          case x: Runtime.Gce =>
+            checkGceRuntime(x, isDryRun)
+          case x: Runtime.AzureVM =>
+            checkAzureRuntime(x, isDryRun)
+        }
+
+      private def checkDataprocCluster(runtime: Runtime.Dataproc, isDryRun: Boolean)(implicit
```

Comment:
These methods seem the same as `DeletedRuntimeChecker`.. wonder if it could be made more DRY?

Reply:
I thought about it too, but the decoupling made is so easy to  develop and test that I ended up liking the framework and kept it, but if you feel strongly about it, I am happy to refactor